log4j-core/src/main/java/org/apache/logging/log4j/core/config/properties/PropertiesConfigurationBuilder.java

```diff
@@ -124,6 +125,26 @@ public PropertiesConfiguration build() {
             }
         }
 
+        Properties monitorResources = PropertiesUtil.extractSubset(rootProperties, "monitorResources");
+        if (monitorResources.size() > 0) {
+            final String monitorResourcesType = (String) monitorResources.remove("type");
+            if (!"MonitorResources".equals(monitorResourcesType)) {
+                throw new ConfigurationException(
+                        "No or invalid type provided for monitorResouces - must be MonitorResources");
+            }
+            final Map<String, Properties> monitorResourceMap =
+                    PropertiesUtil.partitionOnCommonPrefixes(monitorResources);
+            for (final Map.Entry<String, Properties> entry : monitorResourceMap.entrySet()) {
+                final Properties monitorResourceProps = entry.getValue();
+                final String monitorResourceType = (String) monitorResourceProps.remove("type");
+                if (!"MonitorResource".equals(monitorResourceType)) {
+                    throw new ConfigurationException(
+                            "No or invalid type provided for monitorResouce - must be MonitorResource");
+                }
+                builder.add(createMonitorResource(entry.getKey().trim(), entry.getValue()));
+            }
+        }
+
```

Comment:
2. Add an overload of `processRemainingProperties` that accepts a `ConfigurationBuilder`:
    ```java
    private void processRemainingProperties(ConfigurationBuilder builder, Properties properties);
    ```
3. Invoke this new method at the end of the `build()` method, replacing the less generic implementation above.
4. One remaining issue: the `build()` method contains multiple `Properties.getProperty()` calls, which do not remove entries from the `Properties` object.
To prevent duplicate processing in `processRemainingProperties`, replace these calls with `Properties.remove()`.